src/material-experimental/mdc-list/list-base.ts

```diff
@@ -77,20 +76,133 @@ export abstract class MatListItemBase implements AfterContentInit, OnDestroy, Ri
+
+  protected _adapter: MDCListAdapter = {
+    getListItemCount: () => this._items.length,
+    listItemAtIndexHasClass:
+        (index, className) => this._elementAtIndex(index).classList.contains(className),
+    addClassForElementIndex:
+        (index, className) => this._elementAtIndex(index).classList.add(className),
+    removeClassForElementIndex:
+        (index, className) => this._elementAtIndex(index).classList.remove(className),
+    getAttributeForElementIndex: (index, attr) => this._elementAtIndex(index).getAttribute(attr),
+    setAttributeForElementIndex:
+        (index, attr, value) => this._elementAtIndex(index).setAttribute(attr, value),
+    getFocusedElementIndex: () => this._indexForElement(this._document?.activeElement),
+    isFocusInsideList: () => this._element.nativeElement.contains(this._document?.activeElement),
+    isRootFocused: () => this._element.nativeElement === this._document?.activeElement,
+    focusItemAtIndex: index =>  this._elementAtIndex(index).focus(),
+
+    // MDC uses this method to disable focusable children of list items. However, we believe that
+    // this is not an accessible pattern and should be avoided, therefore we intentionally do not
+    // implement this method. In addition, implementing this would require violating Angular
+    // Material's general principle of not having components modify DOM elements they do not own.
+    // A user who feels they really need this feature can simply listen to the `(focus)` and
+    // `(blur)` events on the list item and enable/disable focus on the children themselves as
+    // appropriate.
+    setTabIndexForListItemChildren: () => {},
+
+    // The following methods have a dummy implementation in the base class because they are only
+    // applicable to certain types of lists. They should be implemented for the concrete classes
+    // where they are applicable.
+    hasCheckboxAtIndex: () => false,
+    hasRadioAtIndex: () => false,
+    setCheckedCheckboxOrRadioAtIndex: () => {},
+    isCheckboxCheckedAtIndex: () => false,
+
+    // TODO(mmalerba): Determine if we need to implement these.
+    getPrimaryTextAtIndex: () => '',
+    notifyAction: () => {},
+  };
+
+  protected _foundation: MDCListFoundation;
+
+  protected _document: Document;
+
+  private _itemsArr: MatListItemBase[] = [];
+
+  private _subscriptions = new Subscription();
+
+  constructor(protected _element: ElementRef<HTMLElement>, @Inject(DOCUMENT) document: any) {
+    super();
+    this._document = document;
+    this._isNonInteractive = false;
+    this._foundation = new MDCListFoundation(this._adapter);
+  }
+
+  ngAfterViewInit() {
+    this._subscriptions.add(
+        this._items.changes.pipe(startWith(null))
+            .subscribe(() => this._itemsArr = this._items.toArray()));
+    this._foundation.init();
+    for (let i = 0; this._items.length; i++) {
+      this._elementAtIndex(i).tabIndex = i === 0 ? 0 : -1;
```

Comment:
Could you add a comment on why we need this? Also I was thinking that we can rely on the tabindex being set by individual list-options, or list-items on anchors or buttons. This matches what we do in the non-MDC one right?

Reply:
+1, I'd ideally move this `tabIndex` bit to its own function 
(my preference is to always have lifecycle methods call out to more descriptively named methods unless they're super simple/trivial)

Reply:
The reason its done this way is because the list-items may be used as part of a `<mat-list>` which is non-interactive and doesn't need the MDC foundation. Though I could probably move the logic to the items and just have them check if they're part of an interactive list before setting it

Reply:
Split the logic out into separate methods

Reply:
I'd probably prefer doing it from within the items, as that will make tabindex and component code more self-contained and predictable, but we could revisit in a follow-up .